CdIts.Caffoa.Cli/Parser/ObjectParser.cs

```diff
@@ -60,20 +60,28 @@ private PropertyData ParseProperty(string name, OpenApiSchema schema, bool requi
             if (schema.Properties.Count > 0)
                 throw new CaffoaParserError(
                     "object with properties and additional properties are currently not supported.");
-            property.TypeName = schema.AdditionalProperties.TypeName();
+            if (schema.AdditionalProperties.Reference != null 
+                && schema.AdditionalProperties.Reference.IsLocal)
+            {
+                property.TypeName = schema.AdditionalProperties.Reference.Id.FirstCharUpper();
```

Comment:
please use `schema.AdditionalProperties.Reference.Name()` (using the extension method). That handles the naming in the same way as other elements. Additionally, you do not need the `IsLocal`, as other places handle references to other files.

Reply:
more specific. use `_classNameFunction(schema.AdditionalProperties.Reference.Name())` to Add possible prefix, suffix and handle PascalCase

Reply:
Thanks, that helps a lot